gdal/ogr/ogrsf_frmts/cad/ogrcadlayer.cpp

```diff
@@ -0,0 +1,639 @@
+        for ( std::vector<std::string>::const_iterator
+              iter = poCADGeometry->getEED().cbegin();
+              iter != poCADGeometry->getEED().cend(); ++iter )
+        {
+            sEEDAsOneString += *iter;
+            sEEDAsOneString += ' ';
+        }
+
+        poFeature->SetField( "extentity_data", sEEDAsOneString.c_str() );
+    }
+
+    RGBColor stRGB = poCADGeometry->getColor();
+    GIntBig adRGB[3] { stRGB.R, stRGB.G, stRGB.B };
+    poFeature->SetField( "color", 3, adRGB);
+
+    std::stringstream oStringStream;
+    oStringStream << "PEN(c:#" << std::hex << adRGB[0] << adRGB[1] << adRGB[2];
+    oStringStream << ",w:5px)" << std::dec;
+    poFeature->SetStyleString( oStringStream.str().c_str() );
+
+    std::vector< CADAttrib > oBlockAttrs = poCADGeometry->getBlockAttributes();
+    for( std::vector< CADAttrib >::const_iterator citerBlockAttrs = oBlockAttrs.cbegin(); 
+         citerBlockAttrs != oBlockAttrs.cend(); ++citerBlockAttrs )
+    {
+        for( std::vector< std::string >::const_iterator citerFeatAttrs = asFeaturesAttributes.cbegin(); 
+            citerFeatAttrs != asFeaturesAttributes.cend(); 
+            ++citerFeatAttrs)
+        {
+            if( citerBlockAttrs->getTag() == *citerFeatAttrs )
+            {
+                poFeature->SetField( (*citerFeatAttrs).c_str(), citerBlockAttrs->getTextValue().c_str() );
+            }
+        }
+    }
+
+    switch( poCADGeometry->getType() )
+    {
+        case CADGeometry::POINT:
+        {
+            CADPoint3D * const poCADPoint = ( CADPoint3D* ) poCADGeometry;
+            CADVector stPositionVector = poCADPoint->getPosition();
+
+            poFeature->SetGeometryDirectly( new OGRPoint( stPositionVector.getX(),
+                                                          stPositionVector.getY(),
+                                                          stPositionVector.getZ() ) );
+            poFeature->SetField( "cadgeom_type", "CADPoint" );
+            break;
+        }
+
+        case CADGeometry::LINE:
+        {
+            CADLine * const poCADLine = ( CADLine* ) poCADGeometry;
+            OGRLineString *poLS = new OGRLineString();
+            poLS->addPoint( poCADLine->getStart().getPosition().getX(),
+                           poCADLine->getStart().getPosition().getY(),
+                           poCADLine->getStart().getPosition().getZ() );
+            poLS->addPoint( poCADLine->getEnd().getPosition().getX(),
+                           poCADLine->getEnd().getPosition().getY(),
+                           poCADLine->getEnd().getPosition().getZ() );
+
+            poFeature->SetGeometryDirectly( poLS );
+            poFeature->SetField( "cadgeom_type", "CADLine" );
+            break;
+        }
+
+        case CADGeometry::CIRCLE:
+        {
+            CADCircle * const poCADCircle = ( CADCircle* ) poCADGeometry;
+            OGRGeometry *poCircle = OGRGeometryFactory::approximateArcAngles(
```

Comment:
You can find more information here : https://trac.osgeo.org/gdal/wiki/rfc49_curve_geometries . And for full circle : https://lists.osgeo.org/pipermail/gdal-dev/2015-January/040811.html . ogr/ogrsf_frmts/ili/ili2reader.cpp or ogr/gml2ogrgeometry.cpp can also be looked at.